cimgen/cimgen.py

```diff
@@ -280,6 +281,12 @@ def is_a_float_class(self):
                 return False
         return True
 
+    def is_a_primitive(self):
+        return self.stereotype == "Primitive"
+
+    def is_a_cim_datatype(self):
+        return self.stereotype == "CIMDatatype"
```

Comment:
Adapt it to "is_an_enum_class" to make it more systematic. The same for attributes. So you see immedialely if it is for classes or for attributes.
```suggestion
    def is_a_primitive_class(self):
        return self.stereotype == "Primitive"

    def is_a_datatype_class(self):
        return self.stereotype == "CIMDatatype"
```
To prepare the merging of our branches I tried to nearer my changes in PR #40 to your changes. I included your "CIMDatatype" changes and removed "is_a_float_class" and "is_primitive_float_attribute" which are now not necessary any more. I had to adapt some lang_pack.py (see commit 0fc5750fbca5c7b62cbdb1b9902154123dfecd29). And I also changed your "is_a_cim_datatype" to "is_a_datatype_class" (see above).

There are now 4 disjoint class types in cimgen.py:
1. is_a_datatype_class (= float)
2. is_an_enum_class
3. is_a_primitive_class (Integer, Boolean, Float/Decimal = float, String/Date/DateTime/MonthDay = string)
4. all others are normal classes

and 5 disjoint attribute types:
1. is_class_attribute
2. is_enum_attribute
3. is_datatype_attribute
4. is_list_attribute
6. is_primitive_attribute

How to handle these types is mostly language dependend, e.g. which of the primitive types are implemented by fixed classes or by templates. That's why it makes no sense to define subtypes like primitive_float or primitive_string for class/attribute in cimgen.py. That should be done in the lang_pack.py if necessary.